src/User32/User32.cs

```diff
@@ -2921,6 +2921,381 @@ public static partial class User32
         [DllImport(nameof(User32))]
         public static extern unsafe void mouse_event(mouse_eventFlags dwFlags, int dx, int dy, int dwData, void* dwExtraInfo);
 
+        /// <summary>
+        /// Calculates the required size of the window rectangle, based on the desired size of the client rectangle and the provided DPI. This window rectangle can then be passed to the CreateWindowEx function to create a window with a client area of the desired size.
+        /// </summary>
+        /// <param name="lpRect">A pointer to a <see cref="RECT"/> structure that contains the coordinates of the top-left and bottom-right corners of the desired client area. When the function returns, the structure contains the coordinates of the top-left and bottom-right corners of the window to accommodate the desired client area.</param>
+        /// <param name="dwStyle">The Window Style of the window whose required size is to be calculated. Note that you cannot specify the <see cref="WindowStyles.WS_OVERLAPPED"/> style.</param>
+        /// <param name="bMenu">Indicates whether the window has a menu.</param>
+        /// <param name="dwExStyle">The Extended Window Style of the window whose required size is to be calculated.</param>
+        /// <param name="dpi">The DPI to use for scaling.</param>
+        /// <returns>
+        /// If the function succeeds, the return value is true.
+        /// If the function fails, the return value is false. To get extended error information, call <see cref="GetLastError"/>.
+        /// </returns>
+        [DllImport(nameof(User32), SetLastError = true, CharSet = CharSet.Unicode)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern unsafe bool AdjustWindowRectExForDpi(
+            [In][Out] RECT* lpRect,
```

Comment:
We don't specify `[In]` (or `[Out]`) on virtually any of our parameters. `[In]` seems implied and redundant (I believe in C# it strictly is redundant) and `[Out]` has no effect either AFAIK. Am I wrong?
Unless there's a particular reason to specify them, please remove them.

Reply:
You are right, in my usage, these are all redundant. These can occasionally be useful though. for e.g., [In] ref STRUCT would change the marshaling semantics from being [in, out] to [in] only, which can be useful if the underlying native function works that way. It's just a reflexive habit - whenever I see [in], [out] annotations in native API definition, I tend to transcribe them faithfully in my P/Invoke definitions. Removing.